plugin/storage/grpc/shared/grpc_server.go

```diff
@@ -27,7 +29,9 @@ const spanBatchSize = 1000
 
 // grpcServer implements shared.StoragePlugin and reads/writes spans and dependencies
 type grpcServer struct {
-	Impl StoragePlugin
+	Impl             StoragePlugin
+	ArchiveImpl      ArchiveStoragePlugin
+	CapabilitiesImpl PluginCapabilities
```

Comment:
I was thinking this can be implemented differently: have the single `Impl` that can optionally implement `ArchiveStoragePlugin`, and implement capabilities directly in this server type by checking if `Impl` implements `PluginCapabilities`. This seems like an easier thing to implement for people. For example, the `examples/memstore-plugin/main.go` would not need to change at all.

Reply:
Do you mean to change `Impl` declaration to `interface{}`? 

In other hands having `Impl StoragePlugin` attribute silently implementing inside numerous other interfaces is not transparent architecture as for me.